spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -268,27 +270,8 @@ private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobPar
 					}
 				}
 				if (!existsFlag) {
-					String param;
-					if (boot3Version.equals(schemaTarget)) {
-						if (JobParameter.ParameterType.LONG.equals(jobParametersMap.get(key).getType())) {
-							type = Long.class.getCanonicalName();
-						} else if (JobParameter.ParameterType.DATE.equals(jobParametersMap.get(key).getType())) {
-							type = Date.class.getCanonicalName();
-						} else if (JobParameter.ParameterType.DOUBLE.equals(jobParametersMap.get(key).getType())) {
-							type = Double.class.getCanonicalName();
-						} else if (JobParameter.ParameterType.STRING.equals(jobParametersMap.get(key).getType())) {
-							type = String.class.getCanonicalName();
-						} else  {
-							throw new IllegalArgumentException("Unable to convert " +
-								jobParametersMap.get(key).getType() + " to known type of JobParameters");
-						}
-						param = String.format("%s=%s,%s", key, jobParametersMap.get(key).getValue(), type);
-					} else {
-						param = String.format("%s(%s)=%s", key,
-							jobParametersMap.get(key).getType().toString().toLowerCase(),
-							jobParameters.getString(key));
-					}
-					result.add(param);
+					type = jobParametersMap.get(key).getType().getCanonicalName();
```

Comment:
I notice the key was `toLowerCase`ed before, do we still need to do that?

Reply:
Lower case was for Batch 4 job params types .  For Boot3 we use the class name as the type for job params